libcontainer/container_linux.go

```diff
@@ -1340,7 +1340,7 @@ func (c *Container) makeCriuRestoreMountpoints(m *configs.Mount) error {
 		// The prepareBindMount() function checks if source
 		// exists. So it cannot be used for other filesystem types.
 		// TODO: pass srcFD? Not sure if criu is impacted by issue #2484.
-		if err := prepareBindMount(mountEntry{Mount: m}, c.config.Rootfs); err != nil {
+		if err := prepareBindMount(mountEntry{Mount: m, srcFD: -1}, c.config.Rootfs); err != nil {
```

Comment:
What I don't like about this approach is, with the string we can use the default value ("") to see there's no srcfd. With int, we have to use `-1` everywhere, which is kind of ugly.

I'm on the verge of stating that in all the usage scenarios we can't possibly have `0` for `srcFD`, and thus use `0` as "no srcfd is set". This will simplify the code (but break it, if we'll ever have `0` as srcid).

Reply:
what about `*uint`?  `-1` is the same to `""` , at least to me.

Reply:
I'm all in for *uint or *int, that was how I wrote the srcFD code initially some years ago. But @kolyshkin insisted on using `int`.

IMHO, it should be one `fd` field in the entry of type `*int`.